diff --git a/node_modules/react-native-gesture-handler/android/src/main/jni/cpp-adapter.cpp b/node_modules/react-native-gesture-handler/android/src/main/jni/cpp-adapter.cpp
index 9fc5766..6639b8c 100644
--- a/node_modules/react-native-gesture-handler/android/src/main/jni/cpp-adapter.cpp
+++ b/node_modules/react-native-gesture-handler/android/src/main/jni/cpp-adapter.cpp
@@ -19,7 +19,11 @@ void decorateRuntime(jsi::Runtime &runtime) {
                     return jsi::Value::null();
                 }
 
-                auto shadowNode = shadowNodeFromValue(runtime, arguments[0]);
+                auto shadowNodeList = shadowNodeListFromValue(runtime, arguments[0]);
+                if (shadowNodeList == nullptr || shadowNodeList->empty()) {
+                    return jsi::Value::null();
+                }
+                const auto &shadowNode = shadowNodeList->front();
                 bool isFormsStackingContext = shadowNode->getTraits().check(ShadowNodeTraits::FormsStackingContext);
 
                 return jsi::Value(isFormsStackingContext);
diff --git a/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm b/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm
index 4a0ce69..4db5aaa 100644
--- a/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm
+++ b/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm
@@ -99,7 +99,11 @@ void decorateRuntime(jsi::Runtime &runtime)
         if (!arguments[0].isObject()) {
           return jsi::Value::null();
         }
-        auto shadowNode = shadowNodeFromValue(runtime, arguments[0]);
+        auto shadowNodeList = shadowNodeListFromValue(runtime, arguments[0]);
+        if (shadowNodeList == nullptr || shadowNodeList->empty()) {
+          return jsi::Value::null();
+        }
+        const auto &shadowNode = shadowNodeList->front();
         bool isFormsStackingContext = shadowNode->getTraits().check(ShadowNodeTraits::FormsStackingContext);
         return jsi::Value(isFormsStackingContext);
       });
